submissions/TaniaBondarenko/friends-app/app.js

```diff
@@ -0,0 +1,177 @@
+const NUM_OF_FRIENDS = 20;
+const FRIENDS = document.querySelector(".friends");
+const MALE_RADIO_BUTTON = document.getElementById("male");
+const FEMALE_RADIO_BUTTON = document.getElementById("female");
+const ALL_RADIO_BUTTON = document.getElementById("all");
+let sortedFriends;
+let allFriends;
+let isFiltered = false;
+
+function fetchFriends() {
+  const apiUrl = `https://randomuser.me/api/?results=${NUM_OF_FRIENDS}`;
+  fetch(apiUrl)
+    .then(handleErrors)
+    .then((response) => response.json())
+    .then((data) => (allFriends = data.results))
+    .then(() => addFriends(allFriends))
+    .catch(showErrorMessage);
+}
+
+function handleErrors(response) {
+  if (!response.ok) {
+    throw Error(response.statusText);
+  }
+  return response;
+}
+
+function showErrorMessage() {
+  const wholePage = document.body;
+  wholePage.innerHTML = `<div class="errorMessage">Ops...Something went wrong.</div>`;
+}
+
+function addFriends(friendsToBeAdded) {
+  let fragment = document.createDocumentFragment();
+  friendsToBeAdded.forEach((friend) => {
+    let friendCard = document.createElement("div");
+    friendCard.setAttribute("class", "friend_card");
+    friendCard.innerHTML = `<div class="card_wrapper ${friend.gender}"><div class="name ">${`${friend.name.first} ${friend.name.last}`}</div>
+                                <div class="photo"><img src="${friend.picture.large}"></div>
+                                <div class="info_block">
+                                <div class="age">Age ${friend.dob.age}</div>
+                                <div class="place">${friend.location.city}</div>
+                                <div class="email"><a href="mailto:${friend.mail}" class="email_link">${friend.email}</a></div>
+                                <div class="gender">${friend.gender}</div>
+                                </div></div>
+                                `;
+    fragment.appendChild(friendCard);
+  });
+  FRIENDS.appendChild(fragment);
+}
+
+document.querySelector(".sortPanel").addEventListener("click", showSorteredFriends);
+
+function makeContainerEmpty() {
+  let innerText = " ";
+  FRIENDS.innerHTML = innerText.replace(innerText, " ");
```

Comment:
Why you doing this?

Reply:
Because I need to fill in the page with new friends cards and I'm cleaning the previous  shown.

Reply:
Can you please tell me step-by-step all the operations and why they are happening?

Reply:
Please clarify your question, if my answer is not what you're expected. 
This function replaces  current `innerHtml`, which shows all friends cards, with an empty page.
I use it to clean the area before the functions which are used to show the new set of sorted, filtered or searched friends. 
In this case they are shown correctly.
If I don't use this function, the cards, which are sorted or filtered just are added to the end of current friends cards.

Reply:
No problem. As an example, I need you to do something similar to:

```jsx
const callCallback = (someItems, callback) => {
  someItems.forEach(item => {
    if (item !== undefined) {
      callback(item);
    }
  });

  return true;
};
```
And then imagine that you explain to someone what your code does:

1. Function accepts two arguments - callback and array
2. Then it will run method `Array#forEach` which will go thru the array and run my callback
3. In this callback it will check if an item is not `undefined` and if it's not - call `callback` from arguments of the function and pass item of an array into it
4. In the end it will return `true`

So something like this - step-by-step guide what your part of code does. It called [Rubber duck debugging](https://rubberduckdebugging.com/)

Reply:
1. I declare new variable `innerText` and assign an empty string to it.
2. I call a `replace` method on this empty string and replace this empty string with some new empty string `" "`. 🤦‍♀️
Now, I got it. ))))  My rubber duck became very helpful.
I believe, I even don't need this function and I could use `FRIENDS.innerHTML = " ";` instead of it, when it is need.
 Thank you for your patience 🙏